app/services/scholars_archive/records_by_user_group_and_visibility.rb

```diff
@@ -0,0 +1,43 @@
+module ScholarsArchive
+  class RecordsByUserGroupAndVisibility
+    def call(current_user, facet)
+      query_strings = []
+      if current_user.admin?
+        query_strings << admin_search(facet)
+      elsif current_user.guest?
+        current_user.groups.each do |group|
+          query_strings << public_search(facet, group)
+        end
+      else
+        current_user.groups.each do |group|
+          query_strings << public_search(facet, group)
+          query_strings << authenticated_search(facet, group)
+        end
+      end
+      query_strings << other_owned_records(facet, current_user.username)
+      ActiveFedora::SolrService.get( query_strings.join(" OR "), :rows => 1000000 )["response"]["docs"].map { |item| item[facet.key.gsub("sim", "tesim")] }.uniq
```

Comment:
Move this into the search_builder to keep that functionality together

Reply:
Id rather keep this out of the search builder. In hyrax, the search builders generate the query strings, but they query elsewhere. The query happens usually in a service of some kind.

Reply:
https://github.com/samvera/hyrax/blob/21964e1de237c560136b0136738612b258587552/app/services/hyrax/work_query_service.rb

Reply:
https://github.com/samvera/hyrax/blob/21964e1de237c560136b0136738612b258587552/app/services/hyrax/work_query_service.rb#L51

Reply:
So as I follow it, then you should remove `solr_service` from the builder...?

Reply:
Yeah, looks like thats some left over code from a previous iteration. Ill update.